src/user/controller/password.ts

```diff
@@ -0,0 +1,28 @@
+import Controller from '@curveball/controller';
+import { Context } from '@curveball/core';
+import * as UserService from '../service';
+import { Forbidden } from '@curveball/http-errors';
+import { User } from '../types';
+import * as privilegeService from '../../privilege/service';
+
+class UserPasswordController extends Controller {
+
+  async put(ctx: Context) {
+
+    if (!await privilegeService.hasPrivilege(ctx, 'admin')) {
+      throw new Forbidden('Only users with the "admin" privilege may create new users');
+    }
+
+    const userBody: any = ctx.request.body;
+    const user = await UserService.findById(parseInt(ctx.params.id, 10)) as User;
```

Comment:
Last thing I think you missed. `findById` can return type `User`, `App` or `Group`.

By casting to `User` you are basically saying: "I am going to pretend I got a User back", even if you didn't. Don't cast, and instead throw (for example) a `Forbidden` error, because you should not be able to set a password for Apps or Groups.